docs/developer-docs/setup/hello-world.mdx

```diff
@@ -0,0 +1,254 @@
+nvm install 18
+```
+
+## Start the local replica
+
+```bash
+dfx start
+```
+
+## Create a new project
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```bash
+dfx new hello
+cd hello
+npm install
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```bash
+dfx new hello --type=rust
+cd hello
+npm install
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```bash
+npx azle new azle_hello
+cd hello
+npm install
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+Create the project directory and file structure:
+
+```bash
+mkdir kybra_hello
+cd kybra_hello
+
+mkdir src
+
+touch src/main.py
+touch dfx.json
+```
+
+Create a virtual environment:
+
+```bash
+~/.pyenv/versions/3.10.7/bin/python -m venv venv
+source venv/bin/activate
+```
+
+Install Kybra:
+
+```bash
+pip install kybra
+```
+
+</TabItem>
+</Tabs>
+
+For information about project structure and default configuration, [learn more about deploying your first canister](first-canister.mdx).
+ 
+## Smart contract code
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+// src/hello_backend/main.mo
+actor {
+public query func greet(name : Text) : async Text {
+    return "Hello, " # name # "!";
+};
+};
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+// src/hello_world_backend/src/lib.rs
+#[ic_cdk::query]
+fn greet(name: String) -> String {
+    format!("Hello, {}!", name)
+}
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```typescript
+// azle_hello/src/index.ts
+import { Canister, query, text, update, Void } from 'azle';
+
+// This is a global variable that is stored on the heap
+let message = '';
+
+export default Canister({
+    // Query calls complete quickly because they do not go through consensus
+    getMessage: query([], text, () => {
+        return message;
+    }),
+    // Update calls take a few seconds to complete
+    // This is because they persist state changes and go through consensus
+    setMessage: update([text], Void, (newMessage) => {
+        message = newMessage; // This change will be persisted
+    })
+});
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+Create the project directory and file structure:
+
+```python
+# kybra_hello/src/main.py
+
+from kybra import query, update, void
+
+# This is a global variable that is stored on the heap
+message: str = ''
+
+# Query calls complete quickly because they do not go through consensus
+@query
+def get_message() -> str:
+    return message
+
+# Update calls take a few seconds to complete
+# This is because they persist state changes and go through consensus
+@update
+def set_message(new_message: str) -> void:
+    global message
+    message = new_message # This change will be persisted
+```
+
+</TabItem>
+</Tabs>
+
+
+## Deploy locally 
+
+Deploy the project's canisters locally:
+
+```
+dfx deploy
+```
```

Comment:
This command does many things behind the scene:
1. It sets up a special `cycle-wallet` smart contract for storing cycles. This is needed because all operations require payments in cycles. In a local testnet this wallet comes prefilled with cycles.
2. It build the backend and frontend smart contracts into Wasm binaries.
3. It creates smart contracts on the local testnet and saves their addresses in `./.dfx/local/canister_ids.json`.
4. It installs the Wasm binaries onto the created smart contracts.